falcon/testing/client.py

```diff
@@ -318,6 +323,11 @@ def simulate_request(app, method='GET', path='/', query_string=None,
                 prefix=False,
             )
 
+        if json is not None:
+            body = util.json.dumps(json)
```

Comment:
Should probably set ``ensure_ascii=False`` here.

Reply:
Is that to better test the framework support for request payload encoding?

And in this case I guess one also needs to ensure UTF-8 in the output, and make sure request ``Content-Type`` specifies ``charset=utf-8``?